rubin_sim/scheduler/example/example_scheduler.py

```diff
@@ -0,0 +1,1496 @@
+        else:
+            time_needed = times_needed[1]
+        bfs.append((bf.TimeToTwilightBasisFunction(time_needed=time_needed), 0.0))
+        bfs.append((bf.NotTwilightBasisFunction(), 0.0))
+        bfs.append((bf.PlanetMaskBasisFunction(nside=nside), 0.0))
+        bfs.append((bf.AfterEveningTwiBasisFunction(time_after=time_after_twi), 0.0))
+        # XXX--move kwargs up
+        bfs.append((bf.HaMaskBasisFunction(ha_min=HA_min, ha_max=HA_max), 0.0))
+        # don't execute every night
+        bfs.append((bf.NightModuloBasisFunction(night_pattern), 0.0))
+
+        # unpack the basis functions and weights
+        weights = [val[1] for val in bfs]
+        basis_functions = [val[0] for val in bfs]
+        if filtername2 is None:
+            survey_name = "blob_long, %s" % filtername
+        else:
+            survey_name = "blob_long, %s%s" % (filtername, filtername2)
+        if filtername2 is not None:
+            detailer_list.append(detailers.TakeAsPairsDetailer(filtername=filtername2))
+
+        if u_nexp1:
+            detailer_list.append(detailers.FilterNexp(filtername="u", nexp=1))
+        surveys.append(
+            BlobSurvey(
+                basis_functions,
+                weights,
+                filtername1=filtername,
+                filtername2=filtername2,
+                exptime=exptime,
+                ideal_pair_time=pair_time,
+                survey_note=survey_name,
+                ignore_obs=ignore_obs,
+                nexp=nexp,
+                detailers=detailer_list,
+                **BlobSurvey_params
+            )
+        )
+
+    return surveys
+
+
+def gen_long_gaps_survey(
+    footprints,
+    nside=32,
+    night_pattern=[True, True],
+    gap_range=[2, 7],
+    HA_min=12,
+    HA_max=24 - 3.5,
+    time_after_twi=120,
+):
+    """
+    Paramterers
+    -----------
+    HA_min(_max) : float
+        The hour angle limits passed to the initial blob scheduler (hours).
+    gap_range : list of float ([2, 7])
+        The limits of the target time gaps (hours)
+    time_after_twi : float (120)
+        The amount of time that should have passed after twilight to start
+        trying to observe long gaps (minutes).
+    """
+
+    surveys = []
+    f1 = ["g", "r", "i"]
+    f2 = ["r", "i", "z"]
+    # Maybe force scripted to not go in twilight?
+
+    for filtername1, filtername2 in zip(f1, f2):
+        blob = blob_for_long(
+            footprints=footprints,
+            nside=nside,
+            filter1s=[filtername1],
+            filter2s=[filtername2],
+            night_pattern=night_pattern,
+            time_after_twi=time_after_twi,
+            HA_min=HA_min,
+            HA_max=HA_max,
+        )
+        scripted = ScriptedSurvey([], nside=nside, ignore_obs=["blob", "DDF", "twi"])
+        surveys.append(
+            LongGapSurvey(blob[0], scripted, gap_range=gap_range, avoid_zenith=True)
+        )
+
+    return surveys
+
+
+def gen_GreedySurveys(
```

Comment:
Maybe call this get_GreedySuveys? Using caps suggests to me that it is defining a new class (or list of new classes), rather than creating a list of instances of existing classes.